tests/pipeline/test_compiler.py

```diff
@@ -168,6 +169,7 @@ def test_docker_compiler(setup_pipeline, tmp_path_factory):
     example_dir, pipeline, _ = setup_pipeline
     compiler = DockerCompiler()
     with tmp_path_factory.mktemp("temp") as fn:
+        pipeline.base_path = str(fn)
```

Comment:
Not a big fan of this. Can we turn the fixture into a factory function?

See [this example from connexion](https://github.com/spec-first/connexion/blob/f8f461c5240f6d01b0d8c7104387993ce811b2c7/tests/conftest.py#L75).

Reply:
If it's fine for you I would exclude it here. I'll create an separat issue for this. I took a look into the test and I think it would result in a bigger refactoring. It would make indeed sense to have a pipeline factory cause it is needed in several test cases. Not only here in this file.

Reply:
We already have code like this that we could reuse more.

eg: https://github.com/ml6team/fondant/blob/7ec04dcbe3e2aa6a63e91d28e201bbff35304abf/tests/pipeline/test_compiler.py#L132

This is one is even parameterized so you can easily run the tests on multiple pipelines.